File: apps/api/app/schemas/cost.py
"""Cost tracking schemas — response models for cost and token usage.

FM-047: Pydantic models for cost records and summaries.
"""

import uuid
from datetime import datetime

from pydantic import BaseModel


class CostRecordRead(BaseModel):
    id: uuid.UUID
    model_name: str
    prompt_tokens: int
    completion_tokens: int
    total_tokens: int
    cost_usd: float
    project_id: uuid.UUID | None
    run_id: uuid.UUID | None
    task_id: uuid.UUID | None
    caller: str
    created_at: datetime

    model_config = {"from_attributes": True}


class CostRecordList(BaseModel):
    items: list[CostRecordRead]
    total: int

File: apps/web/types/cost.ts
/** Cost tracking types matching the backend schemas (FM-047/cost tracking). */

export interface CostRecord {
  id: string;
  model_name: string;
  prompt_tokens: number;
  completion_tokens: number;
  total_tokens: number;
  cost_usd: number;
  project_id: string | null;
  run_id: string | null;
  task_id: string | null;
  caller: string;
  created_at: string;
}

export interface CostRecordList {
  items: CostRecord[];
  total: number;
}

export interface CostSummary {
  total_cost_usd: number;
  total_tokens: number;
  record_count: number;
  by_model: Record<string, { cost_usd: number; tokens: number; count: number }>;
}
